rust/pact_ffi/tests/tests.rs

```diff
@@ -1182,6 +1183,405 @@ fn repeated_interaction() {
+  pactffi_with_specification(pact_handle, PactSpecification::V4);
+  let description = CString::new("description 2").unwrap();
+  let content_type = CString::new("application/json").unwrap();
+  let request_body_with_matchers = CString::new("{\"id\": {\"value\":\"1\",\"pact:matcher:type\":\"integer\"}}").unwrap();
+  let interaction_handle = pactffi_new_message_interaction(pact_handle, description.as_ptr());
+  let body_bytes = request_body_with_matchers;
+  pactffi_with_body(interaction_handle.clone(),InteractionPart::Request, content_type.as_ptr(), body_bytes.as_ptr());
+  let result_3 = pactffi_pact_handle_write_file(pact_handle, tmp_dir.as_ptr(), false);
+  expect!(result_3).to(be_equal_to(0));
+  let pact_file: Option<String> = pact_default_file_name(&pact_handle);
+  pactffi_free_pact_handle(pact_handle);
+
+  // end setup new pact
+
+  let pact_path = tmp.path().join(pact_file.unwrap());
+  let f= File::open(pact_path).unwrap();
+
+  let mut json: Value = serde_json::from_reader(f).unwrap();
+  json["metadata"]["pactRust"] = Value::Null;
+  assert_eq!(serde_json::to_string_pretty(&json).unwrap(),
+  r#"{
+  "consumer": {
+    "name": "MergingPactC"
+  },
+  "interactions": [
+    {
+      "description": "description 1",
+      "pending": false,
+      "providerStates": [
+        {
+          "name": "state_desc_1",
+          "params": {
+            "id": "1"
+          }
+        }
+      ],
+      "request": {
+        "headers": {
+          "Accept": [
+            "application/json"
+          ]
+        },
+        "method": "GET",
+        "path": "/api/orders/404"
+      },
+      "response": {
+        "status": 200
+      },
+      "type": "Synchronous/HTTP"
+    },
+    {
+      "contents": {
+        "content": {
+          "id": "1"
+        },
+        "contentType": "application/json",
+        "encoded": false
+      },
+      "description": "description 2",
+      "matchingRules": {
+        "body": {
+          "$.id": {
+            "combine": "AND",
+            "matchers": [
+              {
+                "match": "integer"
+              }
+            ]
+          }
+        }
+      },
+      "metadata": {
+        "contentType": "application/json"
+      },
+      "pending": false,
+      "type": "Asynchronous/Messages"
+    }
+  ],
+  "metadata": {
+    "pactRust": null,
+    "pactSpecification": {
+      "version": "4.0"
+    }
+  },
+  "provider": {
+    "name": "MergingPactP"
+  }
+}"#
+  );
+}
+
+// Issue - Should be able to set version of message pact, and write to file containing v4 interactions
+// seems to be a problem setting the version, which defaults to v3
+// pactffi_new_message will accept a MessageHandle over the FFI barrier, but rust typing wont allow us
+// to pass a PactHandle along with pactffi_with_specification
+#[ignore = "require ability to set pact specification version in pactffi_new_message_pact"]
```

Comment:
i probably need to pull this out into a new issue. pact-new uses a new_message_interaction which is replicated in the test above.